modules/olmv1-troubleshooting-rbac-errors-with-preflight-check.adoc

```diff
@@ -0,0 +1,70 @@
+// Module included in the following assemblies:
+//
+// * extensions/ce/olmv1-installing-an-operator-from-a-catalog.adoc
+
+:_mod-docs-content-type: REFERENCE
+
+[id="olmv1-troubleshooting-rbac-errors-with-preflight-check_{context}"]
+= Preflight permissions check for cluster extensions (Technology Preview)
+
+When you try to install an extension, the Operator Controller performs a dry run of the installation process. This dry run verifies that the specified service account can perform all the actions required to install the extension. This includes creating all the Kubernetes objects in the bundle and the role-based access control (RBAC) rules for the roles and bindings defined by the bundle.
+
+:FeatureName: The preflight permissions check for cluster extensions
+include::snippets/technology-preview.adoc[]
+
+If service account is missing any required RBAC rules, the preflight check fails before the actual installation proceeds. If the preflight check fails, the Operator Controller reports the errors in the status conditions of the extension and in the logs of the Operator Controller.
+
+To proceed with the installation, update the roles and bindings to grant the missing permissions to the service account and apply the changes. If there are no errors, the Operator Controller reconciles the updated permissions and completes the installation.
+
+[id="olmv1-preflight-permissions-check-output_{context}"]
+== Example report from the preflight permissions check
+
+The following report indicates that the service account requires the following missing permissions:
+
+* RBAC rules to perform `list` and `watch` actions for the `services` resource in the core API group for the entire cluster
+* RBAC rules to perform `create` actions for `deployments` resources in the `apps` API group for the `pipelines` namespace
+
+You can access the reports from the preflight permissions check in the status conditions of the cluster extension. The `oc describe clusterextension` command prints information about a cluster extension, including the status conditions.
+
+.Example command
+[source,terminal]
+----
+$ oc describe clusterextension <extension_name>
+----
+
+.Example report
+[source,text]
+----
+apiVersion: v1
+items:
+- apiVersion: olm.operatorframework.io/v1
+  kind: ClusterExtension
+...
+Conditions:
+  Type:    Progressing
+  Status:  False
+  Reason:  Retrying
+  Message: pre-authorization failed: service account requires the following permissions to manage cluster extension:
+           Namespace:"" APIGroups:[] Resources:[services] Verbs:[list,watch]
+           Namespace:"pipelines" APIGroups:["apps"] Resources:[deployments] Verbs:[create]
+----
+where:
+
+`Namespace`:: Specifies the scope of the required RBAC rules at the namespace level, for example the `pipelines` namespace. An empty namespace value, `""`, indicates that you must scope the permission to the cluster.
```

Comment:
I'm a little confused about these "where" conditions. Is "Namespace" in the output above going to be replaced with a different value? If so, then I'd consider whether to style that as <namespace> (and same for below options).

If not, then I don't think the "where:" lead-in makes sense.

Reply:
I thought that we needed to have the `where:` as a lead-in for the definition list.

`Namespace` and the other values won't change. The intent is to describe what the parameters define.